shared/js/l10n.js

```diff
@@ -1374,11 +1390,14 @@
   Context.prototype.registerLocales = function(defLocale, available) {
     /* jshint boss:true */
     this.availableLocales = [this.defaultLocale = defLocale];
+    this.localeSources[defLocale] =
+      (available && (defLocale in available)) ? available[defLocale] : 'bundle';
 
     if (available) {
-      for (var i = 0, loc; loc = available[i]; i++) {
+      for (var loc in available) {
```

Comment:
Ah, okay, I get it now. `available` is no longer an array;  it's an object with lang codes as keys and `bundle` or `extra` as values. I'm not sure if I like this change.  Maybe we should have two methods: `registerLocales` for local languages and `registerAdditionalLanguages` for langpacks?